molgenis-data-annotators/src/main/java/org/molgenis/data/annotation/entity/impl/GavinAnnotator.java

```diff
@@ -0,0 +1,351 @@
+import org.molgenis.data.support.EffectsMetaData;
+import org.molgenis.data.vcf.VcfRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import java.util.stream.StreamSupport;
+
+@Configuration
+public class GavinAnnotator
+{
+	public static final String NAME = "GavinAnnotator";
+	public static final String RESOURCE = "gavin";
+	public static final String RESOURCE_ENTITY_NAME = "ccgg";
+
+	public static final String PATHOMAFTHRESHOLD = "PathoMAFThreshold";
+	public static final String MEANPATHOGENICCADDSCORE = "MeanPathogenicCADDScore";
+	public static final String SPEC95THPERCADDTHRESHOLD = "Spec95thPerCADDThreshold";
+	private static final String CATEGORY = "Category";
+
+	public static final String CLASSIFICATION = "Classification";
+	public static final String CONFIDENCE = "Confidence";
+	public static final String REASON = "Reason";
+	public static final String VARIANT_ENTITY = "Variant";
+
+	public static final int CADD_MAXIMUM_THRESHOLD = 25;
+	public static final int CADD_MINIMUM_THRESHOLD = 5;
+	public static final double MAF_THRESHOLD = 0.00474;
+
+	public enum Category
+	{
+		N1, N2, T1, T2, I1, I2, I3, C1, C2, C3, C4, C5
+
+	}
+
+	@Autowired
+	private Entity gavinAnnotatorSettings;
+
+	@Autowired
+	private DataService dataService;
+
+	@Autowired
+	private Resources resources;
+
+	@Bean
+	public EffectsAnnotator gavin()
+	{
+		LinkedList<AttributeMetaData> attributes = new LinkedList<>();
+		DefaultAttributeMetaData classification = new DefaultAttributeMetaData(CLASSIFICATION,
+				MolgenisFieldTypes.FieldTypeEnum.STRING).setDescription(CLASSIFICATION).setLabel(CLASSIFICATION);
+		DefaultAttributeMetaData confidence = new DefaultAttributeMetaData(CONFIDENCE,
+				MolgenisFieldTypes.FieldTypeEnum.STRING).setDescription(CONFIDENCE).setLabel(CONFIDENCE);
+		DefaultAttributeMetaData reason = new DefaultAttributeMetaData(REASON, MolgenisFieldTypes.FieldTypeEnum.STRING)
+				.setDescription(REASON).setLabel(REASON);
+
+		attributes.add(classification);
+		attributes.add(confidence);
+		attributes.add(reason);
+
+		String description = "Please note that this annotator processes the results from a SnpEff annotation\nTherefor it should be used on the result entity rather than the variant entity itself.\nThe corresponding variant entity should also be annotated with CADD and EXaC";
+
+		AnnotatorInfo gavinInfo = AnnotatorInfo.create(AnnotatorInfo.Status.READY,
+				AnnotatorInfo.Type.PATHOGENICITY_ESTIMATE, NAME, description, attributes);
+		EntityAnnotator entityAnnotator = new QueryAnnotatorImpl(RESOURCE, gavinInfo, new GeneNameQueryCreator(),
```

Comment:
Why not a GevinQueryAnnotatorImpl.java?
It will reduce code in the method and make more clear, reusable?